lang/ruby/lib/avro/json_io.rb

```diff
@@ -0,0 +1,286 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'avro/io'
+require 'json'
+
+module Avro
+  module IO
+    # JsonDecoder keeps track of the data and reads it using a given data type.
+    class JsonDecoder
+      attr_accessor :data
+
+      def io_reader(reader)
+        datum = reader.read
+        datum = '""' if datum.empty?
+        data = JSON.parse(datum, {:quirks_mode => true})
+        @data = data
+
+        self
+      end
+
+      def read_null
+        nil
+      end
+
+      def read_boolean
+        data
```

Comment:
It was an oversight. Added her and for null values too. Also added specs for invalid values for all data types.